src/actions/reactions/bookmark.js

```diff
@@ -0,0 +1,41 @@
+import toastr from 'toastr';
+import axios from '../../utils/axiosInstance';
+
+import {
+  BOOKMARK,
+  BOOKMARK_ERROR,
+  BOOKMARK_ON_ALL,
+  UNBOOKMARK,
+  UNBOOKMARK_ERROR,
+  UNBOOKMARK_ON_ALL,
+} from '../types';
+
+export const bookmarkArticle = (slug, articlePage, id) => async (dispatch) => {
+  try {
+    const { data } = await axios.post(`/articles/${slug}/bookmarks`);
+    if (articlePage === 'allArticles') {
+      dispatch({ type: BOOKMARK_ON_ALL, payload: { data, id } });
+      return toastr.success('Bookmarked article successfully');
+    }
+    dispatch({ type: BOOKMARK, payload: data });
+    return toastr.success('Bookmarked article successfully');
+  } catch ({ response }) {
+    const errorMessage = response.data;
+    return dispatch({ type: BOOKMARK_ERROR, payload: errorMessage });
+  }
```

Comment:
Wouldn't it be intuitive to also display a toastr error message when an error occurs?

Reply:
nice observation @omokaro-faith, It will actually, I'll add that 👍